tree/treeplayer/test/regressions.cxx

```diff
@@ -173,3 +175,12 @@ TEST(TTreeReaderRegressions, IndexedFriend)
 
    gSystem->Unlink(fname);
 }
+
+// https://github.com/root-project/root/issues/18066
+TEST(TSelectorDrawRegressions, TernaryOperator)
+{
+   TTree t;
+   t.Fill();
+   t.Draw("(1?1:1)>>h1(12345,0,20)");
```

Comment:
```suggestion
   t.Draw("(1?2:3)>>h1(12345,0,20)");
```
And also check that the operator 'worked' but checking the value in the bin/histo.